service/crt/k8s_secret.go

```diff
@@ -1,32 +1,35 @@
 package crt
 
 import (
+	"fmt"
+
 	"github.com/giantswarm/certificatetpr"
 	microerror "github.com/giantswarm/microkit/error"
 	"k8s.io/client-go/pkg/api/errors"
 	"k8s.io/client-go/pkg/api/v1"
 )
 
 // CreateCertificate saves the certificate as a k8s secret.
-func (s *Service) CreateCertificate(cert certificateSecret) error {
+func (s *Service) CreateCertificate(secret certificateSecret) error {
 	var err error
 
-	secret := &v1.Secret{
+	k8sSecret := &v1.Secret{
 		ObjectMeta: v1.ObjectMeta{
-			Name: cert.CommonName,
+			Name: s.getSecretName(secret.Certificate),
 			Labels: map[string]string{
-				"clusterComponent": cert.ClusterComponent,
+				certificatetpr.ClusterIDLabel: secret.Certificate.Spec.ClusterID,
+				certificatetpr.ComponentLabel: secret.Certificate.Spec.ClusterComponent,
 			},
 		},
 		StringData: map[string]string{
-			"crt": cert.IssueResponse.Certificate,
-			"key": cert.IssueResponse.PrivateKey,
-			"ca":  cert.IssueResponse.IssuingCA,
+			certificatetpr.Crt.String(): secret.IssueResponse.Certificate,
```

Comment:
Without the .String() it errors because certificatetpr.TLSAssetType is a type. Its a bit ugly but I think the .String() is needed.